photoeditor/src/main/java/ja/burhanrashid52/photoeditor/PhotoSaverTask.java

```diff
@@ -0,0 +1,183 @@
+package ja.burhanrashid52.photoeditor;
+
+import android.annotation.SuppressLint;
+import android.graphics.Bitmap;
+import android.graphics.Canvas;
+import android.os.AsyncTask;
+import android.text.TextUtils;
+import android.util.Log;
+import android.view.View;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import java.io.File;
+import java.io.FileOutputStream;
+
+/**
+ * Created by Burhanuddin Rashid on 18/05/21.
+ *
+ * @author <https://github.com/burhanrashid52>
```

Comment:
I'm not sure if we have any tests that execute the saving logic, but this might be a good opportunity to add one.

Reply:
I  research in the past how to achieve this. But could not find a solution for it. The closest thing I found was [Facebook Screenshot testing](https://github.com/facebook/screenshot-tests-for-android) and [this](https://medium.com/airbnb-engineering/better-android-testing-at-airbnb-a77ac9531cab) Airbnb article.

Reply:
👍 . Yea might require a lot of mocking to get this to work.